src/loaders/common.c

```diff
@@ -258,6 +258,51 @@ int libxmp_test_name(uint8 *s, int n)
 	return 0;
 }
 
+int libxmp_copy_name_for_fopen(char *dest, const char *name, int n)
+{
+	int converted_colon = 0;
+	int i;
+
+	/* libxmp_copy_adjust, but make sure the filename won't do anything
+	 * malicious if it's given to fopen. This should only be used on song files.
+	 */
+	if (!strcmp(name, ".") || strstr(name, "..") || name[0] == '\\' || name[0] == '/')
+		return -1;
+
+
+	for (i = 0; i < n - 1; i++) {
+		uint8 t = name[i];
+		if (!t)
+			break;
+
+		/* Reject non-ASCII symbols as they have poorly defined behavior.
+		 */
+		if (t < 32 || t >= 0x7f)
+			return -1;
+
+		/* Reject anything resembling a Windows-style root path. Allow
+		 * converting a single : to / so things like ST-01:samplename work.
+		 */
+		if (t == ':' && !converted_colon) {
```

Comment:
Do you want to check i==1 here? (i.e. you want to catch `c:\something`, yes?)

Reply:
Not necessarily—some console SDKs (and maybe OSes) allow things like `sdcard:/`, `hd0:/`, etc. I think checking for the slash afterward (or maybe also `\0`) is a better way to detect that.

Reply:
I see, OK.

Reply:
I changed this to also check for `\0`, so I think it should be fine now.